src/github3/checks.py

```diff
@@ -0,0 +1,279 @@
+
+    .. CheckSuite Documentation:
+        http://developer.github.com/v3/checks/suites/
+    """
+
+    class_name = 'CheckSuite'
+    CUSTOM_HEADERS = {
+        'Accept': 'application/vnd.github.antiope-preview+json'
+    }
+
+    def _update_attributes(self, suite):
+        # Import here, because a toplevel import causes an import loop
+        from . import pulls, repos
+        self._api = suite['url']
+        # self.base = Base(pull['base'], self)
+        self.status = suite['status']
+        self.conclusion = suite['conclusion']
+        self.head_branch = suite['head_branch']
+        self.head_sha = suite['head_sha']
+        self.before = suite['before']
+        self.after = suite['after']
+        pull_requests = suite.get('pull_requests', [])
+        self.pull_requests = [
+            pulls.ShortPullRequest(p, self) for p in pull_requests
+        ]
+        self.repository = repos.ShortRepository(suite['repository'], self)
+        self.id = suite['id']
+
+    def _repr(self):
+        return '<{s.class_name} [{s.id}:{s.status}]>'.format(s=self)
+
+    @decorators.requires_app_installation_auth
+    def rerequest(self):
+        """Rerequest the check suite.
+
+        :returns:
+            True if successful, False otherwise
+        :rtype:
+            bool
+        """
+        url = self._build_url('rerequest', base_url=self._api)
+        return self._boolean(self._post(
+            url, headers=CheckSuite.CUSTOM_HEADERS), 201, 404)
+
+    @decorators.requires_app_installation_auth
+    def check_runs(self):
+        """Retrieve the check runs for this suite.
+
+        :returns:
+            the check runs for this commit
+        :rtype:
+            :class:`~github3.checks.CheckRun`
+        """
+        url = self._build_url('check-runs', base_url=self._api)
+        return self._iter(-1, url, CheckRun)
+
+
+class CheckRun(models.GitHubCore):
+    """The :class:`CheckRun <CheckRun>` object.
+
+    .. versionadded:: 1.2.0
+
+    Please see GitHub's `CheckRun Documentation`_ for more information.
+
+    .. attribute:: status
+
+        The current status of the check.
+
+    .. attribute:: conclusion
+
+        The final conclusion of the check. If the run has not concluded
+        this will be None.
+
+    .. attribute:: head_sha
+
+        The sha of the commit at the head of the branch checked.
+
+    .. attribute:: name
+
+        The name of the check.
+
+    .. attribute:: started_at
+
+        A :class:`~datetime.datetime` object representing the date and time
+        when this check run started.
+
+    .. attribute:: completed_at
+
+        A :class:`~datetime.datetime` object representing the date and time
+        when this check run completed. If this run is not completed it will
+        be ``None``.
+
+    .. attribute:: pull_requests
+
+        A list of representations of the pull requests the check run belongs to
+        as :class:`~github3.pulls.ShortPullRequest` (this may be empty).
```

Comment:
We have, above, "this may be empty" as a trailing sentence. Perhaps we should, in both cases, have it as something that looks like:

```rst

.. attribute:: pull_requests

    A list of .... :class:`~github3.pulls.ShortPullRequest`.

    .. note::

         This may be empty.

```

Reply:
Also, thinking about the attribute, GitHub has a tendency of later adding endpoints that shadow attributes. I'd be inclined to head off any conflict at the pass and pre-emptively rename these couple of `pull_requests` attributes to `original_pull_requests`. This follows the naming pattern used elsewhere to indicate the difference between:

- This was included in the response that created the object, and
- This is fetching an updated list of X

Reply:
Interesting. I'll read up on this for prior art and history.